### PR TITLE
Add a newline after metadata when initializing scripts with other metadata blocks

### LINKED ISSUE #12499: Initial `uv add --script` produces invalid result with existing metadata section
### Summary

When running `uv add --script <script> <package>` on a script which has an existing non-`script` metadata section and doesn't have the `script` section, uv fails to put a blank line between the sections, resulting in an unintuitive error when running the script or adding more dependencies.

MRE:
```bash
$ cat myscript.py
# /// nonstandardsection
# Hello,
# World!
# ///

print('Hello, world')

$ uv run --script myscript.py 
Hello, world
$ uv add --script myscript.py requests
Updated `myscript.py`
$ cat myscript.py 
# /// script
# requires-python = ">=3.13"
# dependencies = [
#     "requests",
# ]
# ///
# /// nonstandardsection
# Hello,
# World!
# ///

print('Hello, world')

$ uv run --script myscript.py 
error: TOML parse error at line 5, column 1
  |
5 | ///
  | ^
invalid key
$ uv add --script myscript.py fastapi
error: TOML parse error at line 5, column 1
  |
5 | ///
  | ^
invalid key
$ nano myscript.py # add a blank line between the sections
$ cat myscript.py 
# /// script
# requires-python = ">=3.13"
# dependencies = [
#     "requests",
# ]
# ///

# /// nonstandardsection
# Hello,
# World!
# ///

print('Hello, world')

$ uv run --script myscript.py 
Installed 5 packages in 6ms
Hello, world
```

Possibly related: #6366

### Platform

Linux Mint 22.1

### Version

uv 0.6.9

### Python version

Python 3.13.0

### PR DESCRIPTION
**Pull request by merlinz01**
## Summary

uv doesn't separate the metadata block from other blocks when adding the `script` block to a script, which results in the next block being considered part of the script block and causes errors when running.

See #12499 for more details.

Closes #12499

## Test Plan

I manually tested the most common scenario, but there's a few edge cases that would be good to have tests for.

I would have written the tests also, but I was running into errors like this:
```bash
$ cargo test --package uv-scripts
   Compiling uv-configuration v0.0.1 (/home/merlin/Projects/uv/crates/uv-configuration)
error: cannot find attribute `value` in this scope
 --> crates/uv-configuration/src/project_build_backend.rs:8:38
  |
8 |     #[cfg_attr(feature = "schemars", value(hide = true))]
  |                                      ^^^^^

error: could not compile `uv-configuration` (lib) due to 1 previous error
```

### CODE REVIEW

**Comment by charliermarsh** [comment]
Thanks, a few tests would be great. You can apply this patch:

```diff
diff --git a/crates/uv-configuration/src/project_build_backend.rs b/crates/uv-configuration/src/project_build_backend.rs
index fb1a171e9..90fac2ec9 100644
--- a/crates/uv-configuration/src/project_build_backend.rs
+++ b/crates/uv-configuration/src/project_build_backend.rs
@@ -5,7 +5,7 @@
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
 pub enum ProjectBuildBackend {
     #[cfg_attr(feature = "clap", value(hide = true))]
-    #[cfg_attr(feature = "schemars", value(hide = true))]
+    #[cfg_attr(feature = "schemars", schemars(skip))]
     /// Use uv as the project build backend.
     Uv,
     #[default]
```

**Comment by charliermarsh** [comment]
Ah, looks like you also need https://github.com/astral-sh/uv/pull/12504.

**Comment by merlinz01** [comment]
Hmm, now I'm getting lots of errors like this:
```bash
$ cargo test --package uv-scripts
   Compiling uv-settings v0.0.1 (/home/merlin/Projects/uv/crates/uv-settings)
error[E0277]: the trait bound `PythonPreference: ValueEnum` is not satisfied
   --> crates/uv-settings/src/settings.rs:252:35
    |
252 |     pub python_preference: Option<PythonPreference>,
    |                                   ^^^^^^^^^^^^^^^^ the trait `ValueEnum` is not implemented for `PythonPreference`
    |
    = help: the trait `ValueEnum` is implemented for `clap::ColorChoice`
# 25 more similar errors...
```

This is what I'm running:
```bash
$ cargo --version
cargo 1.85.1 (d73d2caf9 2024-12-31)
$ rustup --version
rustup 1.26.0 (2024-04-01)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.85.1 (4eb161250 2025-03-15)`
$ rustc --version
rustc 1.85.1 (4eb161250 2025-03-15)
$ uname -a
Linux myhostname 6.8.0-55-generic #57-Ubuntu SMP PREEMPT_DYNAMIC Wed Feb 12 23:42:21 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/os-release 
NAME="Linux Mint"
VERSION="22.1 (Xia)"
```

**Comment by charliermarsh** [comment]
Those PRs should merge shortly, then you can rebase on top of them if you'd like. (The problem, I think, is that we tend to run `cargo test` over the entire workspace, so as long as one crate enables the necessary features, all crates have access. Thus, you only see these failures if you try to test a crate individually.)

**Comment by merlinz01** [comment]
Thanks for the speedy fix!  I have added tests.

File: crates/uv-scripts/src/lib.rs [hunk 1]
```diff
@@ -782,4 +803,240 @@ mod tests {
...
+                .unwrap();
+        assert_eq!(prelude, "");
+        assert_eq!(
+            metadata.raw,
+            indoc::indoc! {r#"
+            requires-python = ""
+            dependencies = []
+            "#}
+        );
+        assert_eq!(
```
Comment (charliermarsh) [inline crates/uv-scripts/src/lib.rs:1029]: It _might_ be easier to do this via snapshot testing?

Comment (merlinz01) [reply crates/uv-scripts/src/lib.rs:1029]: I can't find how to import the `uv_snapshot` macro.

**Review by charliermarsh — APPROVED** [review]
Thanks.
